### PR TITLE
RC-58: fix crash 8900 that triggers on highlighted objects which are edited at the same time

### PR DESCRIPTION
**Pull request by samcake**
This pr address bug  [8900](https://highfidelity.fogbugz.com/f/cases/8900/interface-crash-when-changing-dimensions-of-circus-hammer)

This was due to the fact that the sub items provided by the meta item of the model entity (of the circus entity) were pointing to wrong ids which were invalid. 
These were the ids used for the highlighting effect, they end up being returning invalid items which would crash down the road.

WE changed the sub renderItemIDs list of the Model entity in its meta through a transaction instead of going through the pointer to the model.
THis garanty that the render item IDs returned from the meta item are the correct one and that they are updated in thread safe way through a copied value from game loop to render loop.

## TEST PLAN
RePlay the scenario of the bug, it shouldn;t crash:
(1) from the marketplace: rez the circus hammer
(2) right-click to get info icon
(3) go into create mode
(4) select the hammer
(5) drag a selection handle as if to scale the hammer
(6) --> NO crash

Big up to @Nex-Pro  for finding the bug with a big hammer
Big up to @SamGondelman for identifying the root cause for the bug

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/4691/HighFidelity-Beta-PR11753-45e5d91b9dcc2cc035b25965d2e16869c39b6c7b.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/4691/HighFidelity-Beta-PR11753-45e5d91b9dcc2cc035b25965d2e16869c39b6c7b.exe

**Comment by Nex-Pro** [comment]
Testing

**Comment by Nex-Pro** [comment]
@samcake Actually I'm pretty sure I found the bug.

**Comment by Nex-Pro** [comment]
@samcake This doesn't crash but the behaviour is changed. 
In earlier builds if you selected create mode while the info highlight was active on the entity the highlighting and the (i) would be dismissed.
Is it desirable to have the info highlight active while editing the entity ?

**Comment by samcake** [comment]
@Nex-Pro I don't know about the behavior you are pointing out.  I doubt it s due to this pr though

**Comment by Nex-Pro** [comment]
@samcake I think it was introduced in the PR that this bug was found in.

**Comment by divyaelango88** [comment]
@Nex-Pro good point, the bug report did not capture the two issues. It only captured the crash and I am guessing this pr fixes the crash - This is a P1. Perhaps we should file another bug report about the change in behavior that PR caused and have that addressed separately as part of another release. What do you think @samcake ?

File: libraries/entities-renderer/src/RenderableModelEntityItem.cpp [hunk 1]
```diff
@@ -953,7 +953,8 @@ ItemKey ModelEntityRenderer::getKey() {
 
 uint32_t ModelEntityRenderer::metaFetchMetaSubItems(ItemIDs& subItems) { 
     if (_model) {
-        auto metaSubItems = _model->fetchRenderItemIDs();
+      //  auto metaSubItems = _model->fetchRenderItemIDs();
```
Comment (ZappoMan) [inline libraries/entities-renderer/src/RenderableModelEntityItem.cpp]: what's this dead code for?

File: libraries/entities-renderer/src/RenderableModelEntityItem.cpp [hunk 2]
```diff
@@ -1202,6 +1203,10 @@ void ModelEntityRenderer::doRenderUpdateSynchronousTyped(const ScenePointer& sce
         if ((bool)model) {
             model->removeFromScene(scene, transaction);
             withWriteLock([&] { _model.reset(); });
+            transaction.updateItem<PayloadProxyInterface>(getRenderItemID(), [](PayloadProxyInterface& data) {
+                auto re = static_cast<EntityRenderer*>(&data);
```
Comment (ZappoMan) [inline libraries/entities-renderer/src/RenderableModelEntityItem.cpp]: coding standard: variable names should be descriptive

File: libraries/entities-renderer/src/RenderableModelEntityItem.cpp [hunk 3]
```diff
@@ -1297,6 +1302,12 @@ void ModelEntityRenderer::doRenderUpdateSynchronousTyped(const ScenePointer& sce
             render::Item::Status::Getters statusGetters;
             makeStatusGetters(entity, statusGetters);
             model->addToScene(scene, transaction, statusGetters);
+
+            auto newRenderItemIDs{ model->fetchRenderItemIDs() };
+            transaction.updateItem<PayloadProxyInterface>(getRenderItemID(), [newRenderItemIDs](PayloadProxyInterface& data) {
+                auto re = static_cast<EntityRenderer*>(&data);
```
Comment (ZappoMan) [inline libraries/entities-renderer/src/RenderableModelEntityItem.cpp]: coding standard

**Comment by cain-kilgore** [comment]
Testing.

**Comment by cain-kilgore** [comment]
# Test Passed
Test | Result
--- | ---
(1) from the marketplace: rez the circus hammer | Pass
(2) right-click to get info icon | Pass
(3) go into create mode | Pass
(4) select the hammer | Pass
(5) drag a selection handle as if to scale the hammer | Pass
(6) --> NO crash | Pass

146dc3ec-d924-4e4b-abcf-51b65955bde8

No crash! :)

###### Intel i5 7600K @ 4Ghz, 16GB DDR4 2133Mhz Corsair Vengeance, Asus Strix GTX 1050Ti 3GB, HTC Vive

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/4696/HighFidelity-Beta-PR11753-0d900d7871ea681bd700a9db083a55913e8ee53b.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/4696/HighFidelity-Beta-PR11753-0d900d7871ea681bd700a9db083a55913e8ee53b.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/4699/HighFidelity-Beta-PR11753-4cc0a4699895784d9582788693a6535cfc2f4114.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/4699/HighFidelity-Beta-PR11753-4cc0a4699895784d9582788693a6535cfc2f4114.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/4702/HighFidelity-Beta-PR11753-dfa4afc7f6410df9a18c6a5242dbae4a3d469795.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/4702/HighFidelity-Beta-PR11753-dfa4afc7f6410df9a18c6a5242dbae4a3d469795.exe

**Comment by cain-kilgore** [comment]
Testing.

**Comment by cain-kilgore** [comment]
# Test Passed
hammer

All good!

**Pull request merged**